clang/include/clang/Basic/DiagnosticParseKinds.td

```diff
@@ -891,6 +891,10 @@ def missing_template_arg_list_after_template_kw : Extension<
   "a template argument list is expected after a name prefixed by the template "
   "keyword">, InGroup<DiagGroup<"missing-template-arg-list-after-template-kw">>,
   DefaultError;
+def warn_missing_template_arg_list_after_template_kw_deprecated : Warning<
+  "the use of the keyword template before the qualified name of a class or "
+  "alias template without a template argument list is deprecated">,
+  InGroup<DiagGroup<"warn-missing-template-arg-list-after-template-kw-deprecated-deprecated">>;
```

Comment:
```suggestion
def warn_missing_template_arg_list_after_template_kw_deprecated : Warning<
  "use of 'template' in a qualified name of a class or alias template without a template argument list is deprecated">,
  InGroup<DiagGroup<"warn-missing-template-arg-list-after-template-kw-deprecated-deprecated">>;
```
This seems like it might be a more succinct way to say the same thing (needs 80 col reformatting, I'm sure).